### ISSUE TITLE
Duplicated mesh faces

### ISSUE THREAD

**Issue by JanFilipec** [IfcOpenShell/IfcOpenShell#4985]
See:
https://community.osarch.org/discussion/2243/how-to-make-a-mesh-manifold#latest

Current version of Blender+Blender BIM produces a non-manifold mesh with duplicate faces.
The problem does not exist in Blender 3.6.1+BBIM 0.0.230902

**Comment by falken10vdl** [comment]
@JanFilipec can you tell us if this is still the case in latest builds of Bonsai?
Thanks!

**Comment by JanFilipec** [comment]
Yes, the problem is still present in bonsai-0.8.3-alpha2504241459 (Blender 4.3.2 Python 3.11 Windows)

**Comment by BIMvoice** [comment]
I tried to reproduce this on current Bonsai (ifcopenshell 0.8.6-alpha). I built synthetic geometry across the likely culprits, extruded solids, boolean cuts, IfcPolygonalFaceSet (closed and open), IfcFacetedBrep, and open surface models (IfcFaceBasedSurfaceModel / IfcShellBasedSurfaceModel, including non-planar faces), and checked the triangle count against the unique face set in every case. No duplicate faces in any of them, including through the full Blender import path (a boolean-cut box gives 32 polygons, 32 unique).

The Bonsai loader maps kernel faces one-to-one to Blender polygons, so it can't double faces on its own, and the OCCT triangulation serializer emits each triangle once. That means the "every face doubled, one up one down" you saw comes either from an older kernel path that has since changed, or from the source file already containing coincident double-sided faces.

Could you attach the IFC file (or a minimal cut-down of it) that shows the duplicated faces? Without it I can't pin down which case you hit. If you still have the file that was fine in Blender 3.6.1, that would help most.
